### PR TITLE
Add Affymetrix analysis modules, observation_name_col, make GTF optional

### PR DESCRIPTION
**Pull request by pinin4fjords**
This PR:

- Adds handling of Affymetrix array data via affy/justrma and limma/differential. There was not a direct analogy with RNA-seq processes, so this entailed some re-wiring.
- Adds separate config profiles for affy and rna-seq, to be used when working with those platforms.
- Adds a new test profile for affymetrix array analysis (requires a merge of https://github.com/nf-core/test-datasets/pull/765)
- Removes the absolute dependence on GTF files for feature annotation
- Adds a parameter for choosing a column from the observations for use in display (`observation_name_col`)
- Improves report: 
  - dendrograms for all matrix variants
  - volcano plots for unadjusted as well as adjusted p values

closes https://github.com/nf-core/differentialabundance/issues/47 and other issues.

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/differentialabundance/tree/master/.github/CONTRIBUTING.md)- [ ] If necessary, also make a PR on the nf-core/differentialabundance _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [x] Make sure your code lints (`nf-core lint`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [x] Usage Documentation in `docs/usage.md` is updated.
- [x] Output Documentation in `docs/output.md` is updated.
- [x] `CHANGELOG.md` is updated.
- [x] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: nextflow_schema.json [hunk 1]
```diff
@@ -10,18 +10,23 @@
...
+                    "default": "study",
+                    "description": "A string to identify results in the output directory",
+                    "help_text": "Also used as an identifier in some processes",
+                    "fa_icon": "fas fa-signature"
+                },
+                "study_type": {
+                    "type": "string",
+                    "default": "rnaseq",
+                    "description": "A string identifying the technology used to produce the data",
+                    "help_text": "Currently only 'rnaseq' may be specified. In future options like 'affy_array' are possible.",
```
Comment (apeltzer) [inline nextflow_schema.json:26]:
```suggestion
                    "help_text": "Currently 'rnaseq' or 'affy_array' may be specified.",
```

**Review by apeltzer — APPROVED** [review]
Only tiny bits (affy_array is now an option), otherwise looks great. Will give this a go with some Nanostring data, lets see!

**Comment by pinin4fjords** [comment]
Thanks @apeltzer ! Will merge after https://github.com/nf-core/modules/pull/2890 and https://github.com/nf-core/test-datasets/pull/765 (+ update to test data pointer here)

**Comment by pinin4fjords** [comment]
@apeltzer thanks, but I was actually still adding a couple of tiny fixes!

**Pull request merged**